python/pyarrow/compute.py

```diff
@@ -294,6 +294,46 @@ def mode(array):
     return call_function("mode", [array])
 
 
+def and_(x1, x2):
```

Comment:
@jorisvandenbossche @pitrou 

AFAIK we can't have methods named `and` and `or` in Python. For now I went with `and_` and `or_` here, for consistency with `and_kleene` and `or_kleene`, but for example numpy uses `np.logical_and` and `np.logical_or`

Reply:
We already have a function named "and":
```python
>>> from pyarrow import compute
>>> getattr(compute, "and")
<function pyarrow.compute.and(left, right, *, memory_pool=None)>
```
So you should just fix `compute.py` so that it's exported as `and_`.

Reply:
Right. Fixed now